### PR TITLE
Add WaylandSource behind the `calloop` cargo feature

### PR DESCRIPTION
**Pull request by hw0lff**
I have added the `WaylandSource` struct from sctk to the wayland-client crate.
It can be enabled with the `calloop` feature flag.
I also wrote a little description for the feature in `src/lib.rs`.

### CODE REVIEW

File: wayland-client/src/calloop.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
+//! Utilities for using an [`EventQueue`] from wayland-client with an event loop that performs polling with
+//! [`calloop`](https://crates.io/crates/calloop).
+
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![warn(missing_docs, missing_debug_implementations)]
+#![forbid(unsafe_op_in_unsafe_fn)]
+#![allow(clippy::new_without_default)]
```
Comment (elinorbgr) [inline wayland-client/src/calloop.rs:7]: Please don't add such attributes in the module like this, they are either redundant with the crate-level ones or have no reason to be here.

File: wayland-client/src/calloop.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+use calloop::{
+    generic::Generic, EventSource, InsertError, Interest, LoopHandle, Mode, Poll, PostAction,
+    Readiness, RegistrationToken, Token, TokenFactory,
+};
+use nix::errno::Errno;
+use wayland_backend::client::{ReadEventsGuard, WaylandError};
+
+/// An adapter to insert an [`EventQueue`] into a calloop [`EventLoop`](calloop::EventLoop).
+///
+/// This type implements [`EventSource`] which generates an event whenever events on the display need to be
```
Comment (elinorbgr) [inline wayland-client/src/calloop.rs:24]:
```suggestion
/// This type implements [`EventSource`] which generates an event whenever events on the event queue need to be
```

**Review by elinorbgr — CHANGES_REQUESTED** [review]
LGTM overall, just a few points to fix.

Also, could you add an entry in the `Unreleased` section of the changelog for wayland-client introducing this change?

File: wayland-client/src/calloop.rs (hunk 1 above)
Comment (hw0lff) [reply wayland-client/src/calloop.rs:7]: Ah, sorry I must have copied the file from the wrong subdirectory. I think this is a leftover from the attempted separate `WaylandSource` crate. I'll remove it!

**Comment by elinorbgr** [comment]
Ok, LGTM, however looks like we have a few CI issues. These are not directly related with your PR, but I plan to fix them before merging.

**Comment by hw0lff** [comment]
What about the renaming of `display` to `event queue`?
Is it only correct in the first instance or all of them?
The topics around it seem to be the same to me.

**Comment by elinorbgr** [comment]
Oh right. Only the changes in the doc comment should be kept.

**Comment by hw0lff** [comment]
Like this [90cbe6f](https://github.com/Smithay/wayland-rs/pull/583/commits/90cbe6ff55cd546abe2608ecbd9f581d537a6475)?

**Comment by elinorbgr** [comment]
Yes 👍

**Review by elinorbgr — APPROVED** [review]

**Pull request merged**